tlatools/org.lamport.tlatools/javacc/tla+.jj

```diff
@@ -730,17 +730,15 @@ if (opArgs.kind != N_OpArgs) { ToolIO.out.println("Bug: not N_OpArgs node"); };
 
   private final String msgStackToString(ParseException e) {
     StringBuffer msg;
-    String name = " *unknown* (error occurred before module header)";
-    if (mn != null) name = mn.toString();
 
     msg = new StringBuffer("***Parse Error***\n");
-    if ( expecting != emptyString ) {
+    if ( !expecting.equals(emptyString) ) {
```

Comment:
This might NPE if `expecting` is null.  Perhaps, `!emptyString.equals(expecting)`.